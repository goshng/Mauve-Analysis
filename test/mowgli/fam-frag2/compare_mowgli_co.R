x <- read.table("compare_mowgli_co.txt")
postscript("compareMowgliCoRecombining.ps", width=10, height=10, horizontal = FALSE, onefile = FALSE, paper = "special")
oldpar <- par (mar=c(5, 5, 0.5, 0.5))
plot (x$V2, x$V3, xlab="Recombination intensity", ylab="Number of recombining gene transfer", cex.lab=2, cex.axis=2)
par(oldpar)
dev.off()
cor(x$V2,x$V3)
cor.test(x$V2,x$V3)

postscript("compareMowgliCoHgt.ps", width=10, height=10, horizontal = FALSE, onefile = FALSE, paper = "special")
oldpar <- par (mar=c(5, 5, 0.5, 0.5))
plot (x$V2, x$V4, xlab="Recombination intensity", ylab="Number of horizontal gene transfer", cex.lab=2, cex.axis=2)
par(oldpar)
dev.off()
cor(x$V2,x$V4)
cor.test(x$V2,x$V4)
