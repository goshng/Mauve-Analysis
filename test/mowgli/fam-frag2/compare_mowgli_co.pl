# perl compare_mowgli_co.pl > compare_mowgli_co.txt

# Files:
# fam-frag2.part.txt
# famid-events.txt
# famid-gene.txt
# rimap.gene.txt

# Events are counted for gene families.
# Recombination intensities are scored for SPY genes.
# Gene families could have multiple SPY genes. 
# 1. Find the list of families and the number recombining events form famid-events.txt
# 2. Find the list of genes and recombination intensity from rimap.gene.txt
# 3. Map gene names to gene family names using famid-gene.txt.
# 4. For each gene in the second list from the step 2, find the gene family, for
# which find the number of recombining events from the first list from the step
# 1. Print gene name, recombination intensity, and the number of recombining
# gene transfer.

# Step 1
my %famidEvent;
my %famidHgtEvent;
open FAMILY, "famid-events.txt";
while (<FAMILY>)
{
  chomp;
  my @e = split /\t/;
  $famidEvent{$e[0]} = $e[4];
  $famidHgtEvent{$e[0]} = $e[3];
}
close FAMILY;

# Step 2
my %geneRI;
open RI, "rimap.gene.txt";
my $l = <RI>;
while (<RI>)
{
  chomp;
  my @e = split /\t/;
  $geneRI{$e[0]} = $e[2];
}
close RI;

# Step 3
my %geneToFamily;
open GENETOFAM, "famid-gene.txt";
while (<GENETOFAM>)
{
  chomp;
  my @e = split /\t/;
  if (exists $geneToFamily{$e[1]})
  {
    my $a = $geneToFamily{$e[1]};
    push @$a, $e[0];
  }
  else
  {
    my @a;
    push @a, $e[0];
    $geneToFamily{$e[1]} = [ @a ];
  }
  
}
close GENETOFAM;

# Step 4
foreach my $g (keys %geneRI)
{
  my $f = $geneToFamily{$g};
  foreach my $fid (@$f)
  {
    if (exists $famidEvent{$fid})
    {
      print "$g\t$geneRI{$g}\t$famidEvent{$fid}\t$famidHgtEvent{$fid}\t$fid\n";
    }
  }
}

