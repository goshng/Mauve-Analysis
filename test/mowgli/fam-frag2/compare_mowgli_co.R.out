null device 
          1 
[1] 0.3906464

	Pearson's product-moment correlation

data:  x$V2 and x$V3 
t = 17.0751, df = 1619, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0 
95 percent confidence interval:
 0.3485891 0.4311337 
sample estimates:
      cor 
0.3906464 

null device 
          1 
[1] 0.03302833

	Pearson's product-moment correlation

data:  x$V2 and x$V4 
t = 1.3297, df = 1619, p-value = 0.1838
alternative hypothesis: true correlation is not equal to 0 
95 percent confidence interval:
 -0.01568415  0.08158440 
sample estimates:
       cor 
0.03302833 

